Library/Homebrew/extend/pathname.rb

```diff
@@ -488,6 +488,26 @@ def zipinfo
                  .encode(Encoding::UTF_8, invalid: :replace)
                  .split("\n")
   end
+
+  sig { returns(T::Boolean) }
+  def core_formula_path?
+    fnmatch?("**/homebrew-core/Formula/**.rb", File::FNM_DOTMATCH)
+  end
+
+  sig { returns(T::Boolean) }
+  def core_cask_path?
+    fnmatch?("**/homebrew-cask/Casks/**.rb", File::FNM_DOTMATCH)
+  end
+
+  sig { returns(T::Boolean) }
+  def core_formula_tap?
+    self == CoreTap.instance.path
+  end
+
+  sig { returns(T::Boolean) }
+  def core_cask_tap?
+    self == CoreCaskTap.instance.path
+  end
```

Comment:
This is much cleaner though I'm always hesitant to add more monkey patches.

Reply:
These methods only seem to be used in this file (I don't see any similar patterns throughout the codebase) but it makes more sense to put them here. Good idea!

Reply:
These may need to be called in https://github.com/Homebrew/brew/blob/0e22d828f0dd9e60550a81fee1d8a28aae892738/Library/Homebrew/cli/named_args.rb#L228 to detect formula, but I haven't read that yet.

Reply:
Agreed that I would probably rather not have these in here, particularly if only needed in edit. Perhaps a `Refinement` would work? https://docs.ruby-lang.org/en/2.4.0/syntax/refinements_rdoc.html

Reply:
Moved to Refinement.